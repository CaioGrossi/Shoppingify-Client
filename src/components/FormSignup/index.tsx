import TextField from 'components/TextField';
import Button from 'components/Button';
import { Wrapper, FormError, FormLoading, FormLink } from 'components/Form';
import React, { useState } from 'react';
import api from 'services/api';
import { signIn } from 'next-auth/client';
import Link from 'next/link';

const FormSignup = () => {
  const [loading, setLoadind] = useState(false);
  const [formError, setFormError] = useState('');
  const [values, setValues] = useState({
    email: '',
    password: '',
    first_name: '',
    last_name: ''
  });

  const handleInput = async (field: string, value: string) => {
    setValues((prev) => ({ ...prev, [field]: value }));
  };

  const handleSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    setLoadind(true);

    try {
      await api.post('auth/register', {
        ...values
      });

      signIn('credentials', {
        email: values.email,
        password: values.password,
        callbackUrl: '/'
      });
    } catch (error) {
      setFormError('a user already have this email');
      setLoadind(false);
    }
  };

  return (
    <Wrapper>
      <form onSubmit={handleSubmit}>
        <h1>Signup</h1>
        {!!formError && <FormError>{formError}</FormError>}

        <TextField
          name="first_name"
          placeholder="First Name"
          type="text"
          label="First Name"
          required
          onInputChange={(value) => handleInput('first_name', value)}
        />
        <TextField
          name="last_name"
          placeholder="Last Name"
          type="text"
          label="Last name"
          required
          onInputChange={(value) => handleInput('last_name', value)}
        />

        <TextField
          name="email"
          placeholder="Email"
          type="email"
          label="Email"
          required
          onInputChange={(value) => handleInput('email', value)}
        />
        <TextField
          name="password"
          placeholder="Password"
          type="password"
          label="Password"
          required
          onInputChange={(value) => handleInput('password', value)}
        />

        <Button type="submit" fullWidth>
          {loading ? <FormLoading /> : <>Signup now</>}
        </Button>
      </form>

      <FormLink>
        Already have an account?{' '}
        <Link href="/sign-in">
          <a>Sign in</a>
        </Link>
      </FormLink>
    </Wrapper>
  );
};

export default FormSignup;
