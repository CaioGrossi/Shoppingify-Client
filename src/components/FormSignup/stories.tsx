import { Story, Meta } from '@storybook/react/types-6-0';
import FormSignup from '.';

export default {
  title: 'FormSignup',
  component: FormSignup
} as Meta;

export const Default: Story = () => <FormSignup />;
