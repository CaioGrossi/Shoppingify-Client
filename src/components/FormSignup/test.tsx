import { render, screen } from '@testing-library/react';

import FormSignup from '.';

describe('<FormSignup />', () => {
  it('should render the form', () => {
    const { container } = render(<FormSignup />);

    expect(screen.getByPlaceholderText(/first name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/last name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /signup now/i })
    ).toBeInTheDocument();

    expect(container.parentElement).toMatchSnapshot();
  });
});
