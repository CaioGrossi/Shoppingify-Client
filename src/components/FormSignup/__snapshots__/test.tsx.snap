// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignup /> should render the form 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 8rem;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 28rem;
}

.c4 {
  font-size: 1.8rem;
}

.c5 {
  width: 100%;
  height: 4.5rem;
  padding: 1rem;
  border: 2px solid #bdbdbd;
  border-radius: 1.2rem;
}

.c5:focus-within {
  box-shadow: 0 0 0.8rem #f9a109;
}

.c5:focus-within .c3 {
  color: #f9a109;
}

.c6 {
  border: none;
  width: 100%;
  height: 100%;
  background-color: inherit;
  font-size: 1.8rem;
}

.c6:focus {
  outline: none;
}

.c7 {
  cursor: pointer;
  font-weight: 600;
  background-color: #f9a109;
  color: white;
  font-size: 2rem;
  border: none;
  border-radius: 12px;
  width: 8.7rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5rem;
  width: 100%;
}

.c0 .c1 {
  margin-bottom: 0.5rem;
}

.c0 form {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 h1 {
  font-size: 2.4rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <h1>
          Signup
        </h1>
        <div
          class="c1 c2"
        >
          <label
            class="c3 c4"
            for="first_name"
          >
            First Name
          </label>
          <div
            class="c5"
          >
            <input
              class="c6"
              id="first_name"
              name="first_name"
              placeholder="First Name"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <label
            class="c3 c4"
            for="last_name"
          >
            Last name
          </label>
          <div
            class="c5"
          >
            <input
              class="c6"
              id="last_name"
              name="last_name"
              placeholder="Last Name"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <label
            class="c3 c4"
            for="email"
          >
            Email
          </label>
          <div
            class="c5"
          >
            <input
              class="c6"
              id="email"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 c2"
        >
          <label
            class="c3 c4"
            for="password"
          >
            Password
          </label>
          <div
            class="c5"
          >
            <input
              class="c6"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="c7"
          type="submit"
        >
          <span>
            Signup now
          </span>
        </button>
      </form>
    </div>
  </div>
</body>
`;
